packages/engine_umbrella/apps/engine/lib/processors/booking_processor.ex

```diff
@@ -54,7 +69,12 @@ defmodule Engine.BookingProcessor do
         )
       end)
 
-    PlanStore.put_plan(%{vehicles: vehicles, booking_ids: booking_ids})
+    PlanStore.put_plan(%{
+      vehicles: vehicles,
+      booking_ids: booking_ids,
+      excluded_booking_ids:
+        Enum.map(excluded, &Map.take(&1, [:id, :failure])) |> Enum.map(&handle_booking_failure/1)
```

Comment:
Since you're doing a calling a function that pattern matches and destructures the map, you do not need to do it with Map.take also
```suggestion
        Enum.map(excluded, &handle_booking_failure/1)
```